### DISCUSSION TITLE
Testing testing testing.... WE NEED YOU

### DISCUSSION THREAD

**Discussion by TheOnlyHolyMoly** [vladmandic/sdnext#1425] [Ideas]
Hello, 

as you all know we are getting new features and backends at light-speed pace from Vlad. The number of compatible platforms and technologies is steadily growing. Accordingly the need for some testing support will grow. 

My thought would be to come up with a structure that allows tech and non-tech fans of Vlad to help with testing in a structured manner. 

In the target state that I have in mind, you would get some predefined test packages that you would just execute on your system and return the test results. Additionally, there would need to be defined some complex workflows across the GUI that would need to be excuted manually in a first place. 

Please tag yourself and add your thoughts if you want to help testing and please tag other people that you know that you believe could be willing to help.

**Comment by Aptronymist** [comment]
I'm in, and I've got both a 3060 and rx590.

**Comment by Centurion-Rome** [comment]
Have you also thought about test automation?

**Reply by vladmandic** [reply]
test automation requires available hardware for testing. and for any kind of decent coverage, it would be a set of tests, not a single test - which means it would run for a while. if someone is willing to sponsor a cloud available system, i might be inclined to write tests.
and even then, it would not be feasible to test on all backends supported by this app (cuda, rocm, directml, ipex, mps).

**Reply by Centurion-Rome** [reply]
Maybe just start small. 
Prepare automated test that "free tester" could just run on their local machines instead of clicking around ;)
So they could make office job and just launch tests parallel.

**Reply by vladmandic** [reply]
yes, that is exactly what @TheOnlyHolyMoly is suggesting and i agree with that.

**Reply by Aptronymist** [reply]
Yeah, that would be the exact kind of thing I would be happy to do anytime, especially since I have both cards, though the AMD is only pci_4x while in this machine.

**Comment by brknsoul** [comment]
I'm up for this;
AMD RX 6600 XT 8Gb, Windows 10, DirectML

For reference, I can get approx 1 image per minute using the following settings;
512 x 768
DPM++ 2M Karras
25 steps
4x_NMKD-Siax_200k Upscale 2x
After Detailer detecting faces using mediapipe_full model.

Note the upscaling is automatically done after generation from within the From Text tab, which the function can be enabled from the Postprocessing tab in Settings.
